README.docker

```diff
@@ -0,0 +1,77 @@
+## Docker
+We have added docker and docker-compose files to this project.
+Below are instruction on how to use them.
+
+### Make sure you have docker and docker-compose installed:
+
+##### Linux
+1. [Install Docker](https://docs.docker.com/engine/installation/)
+2. [Install Docker Compose](https://docs.docker.com/compose/install/)
+
+##### OSX / Windows
+- If you are on Mac, check out [Docker for Mac](https://docs.docker.com/docker-for-mac/)
+- If you are on Windows, check out [Docker for Windows](https://docs.docker.com/docker-for-windows/)
+
+These will install `docker`, `docker-compose`, and `docker-machine` on your machine.
+
+
+###  Get ualbertalib/DMP_roadmap source code
+Clone the repository from github:
+``shell
+git clone https://github.com/ualbertalib/DMP_roadmap.git
+cd DMP_roadmap
+``
+
+### Build docker DMP_roadmap docker image
+``shell
+docker build . -t ualbertalib/dmp_roadmap
+``
+
+### Start docker compose
+``shell
+docker-compose up -d
+``
+
+### Run database setup (this needs to be done only once)
+``shell
+docker-compose run web rake db:setup
```

Comment:
and db:migrate?

Reply:
you only need to run db:migrate if you change tables and want to bring new changes to your database image. To get DMP working you need to run db:setup only once. This is exactly the same procedure as we use on PMPY docker image.

Reply:
db:migrate should be part of the rails app deployment process. Yes to get it working for the first time, you only needs db:setup as the schema.rb is up-to-date. But if there are any changes to the database schema/structure, you will have to run db:migrate to update the database. See Jupiter's deploy.sh: https://github.com/ualbertalib/jupiter/blob/master/deploy

PMPY is a ruby application so it doesn't require db: rake tasks at all.

Reply:
I do not know why you are so opposed to using of .env file. It gives you advantage that you need to set up your DB password in one place instead of chasing environment variables all over your docker-compuse.yml file.

Reply:
No, I'm not opposed to the use of .env file as I mentioned in a later comment. But I'd want to make sure if we use it, we use it properly. If you decide to use env_file, then use it for what it is meant to be, as the single source for all the environment variables instead of setting some in env_file, and others in environments.